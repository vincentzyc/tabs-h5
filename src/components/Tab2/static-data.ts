export const expensesText = `【资费详情】
1、月基本费 39 元，套餐内包含 5GB 国内通用流量， 30GB 爱 奇艺、腾讯视频、优酷视频、西瓜视频、抖音短视频、抖音火 山版、快手、网易云音乐等应用定向流量，全国接听免费，赠 送来电显示、天翼云 盘-免费版、189 邮箱-免费版。
2、订购当月获得 40 元话费，订购立即生效，激活即可使 用，不退返现金，不可抵扣国际业务、港澳台业务通信费和 SP/CP 等代收费用。获得话费激活当月有效，变更退订次月 生效，协议期 1 个月，到期自动退订。
3、激活后立即生效。首月执行过渡期 资费，订购当月套餐月 基本费按日计扣（入网当日到月底）， 费用四舍五入到分， 套餐内容按照对应流量按天折算，向上取整。
4、全国接听免费，赠送来电显示和 189 邮箱。
5、套餐超出资费 （1）国内通用流量：前 167MB 按 0.03 元/MB 收费，达 到 167MB 时，额外赠送用户 857MB 流量（即 5 元/GB）；后 续 仍按上述规则收费（即每超出 1GB 按照 5 元/GB 收费）。 （2）国内通话：0.1 元/分钟。 （3）国内短/彩信：0.1 元/条。 （4）其他业务按照标准资费执行。
6、国内通话和接听免费范围不包括港澳台地区
7、手机上网流量仅限中国大陆境内使用,不含 WLAN(Wi-Fi)上 网
8、套餐内包含的国内通用流量适用于流量不清零规则。
9、可订购各类流量包、语音包及短/彩信包产品
10、上网当月套餐外流量消费 600 元及 600 元的倍数则断 网，次月初自动开通，如果客户当月申请继续使用，则继续 提供服务，按照套外资费收费。套餐外流量消费费用包含按 套餐外资费收取的上网流量费，不含用户订购的定向、闲时 等各种流量包功能费。
11、通过指定入口预存 50 元可获赠 50 元话费。预存话费 50 元立即到账，不限制抵扣范围，不可退现；赠送话费 50 元， 从订购次月起连续 5 个月内每月返还 10 元，赠送话费从返还 之日起 5 个月内有效，赠送话费可结转，可抵扣套餐基本费， 不能用于国际及港澳台通信费和由电信代收的信息服务费。
免流规则
1.头条系应用包含：今日头条、今日头条 lite 版、西 瓜视频、抖音火山版、抖音短视频、皮皮虾、懂车帝、半次 元。
2.百度系应用包含：百度 APP、百度贴吧、百度地图、 百度网盘、百度手机助手、百度百科、秒懂百科、百度知道、百度文库、百度输入法、地图淘金、千千音乐、百度翻译、 百度新闻、百度浏览器、好看视频。 其中：手机百度、百度网盘、百度手机助手、百度新闻、 百度浏览器、好看视频享受免流服务前需先至对应 APP 进行 绑定激活。 绑定入口： （1）手机百度：【个人中心】-【免流量特权】-【活动页面】 -【激活免流】 （2）百度网盘：【更多】-【设置】-【免流量特权】-【活动 页面】-【激活免流】（只限百度网盘超级会员绑定激活免流， 非会员无法免流） （3）百度手机助手：【管理】-【免流量服务】-【活动页面】 -【激活免流】 （4）好看视频：【我的】-【免流特权】-【活动页面】-【激 活免流】 注：以上需绑定激活的应用（除百度网盘 APP），在同一手机 终端使用时，只需激活其中一款应用，其他应用无需绑定激活 即可享受免流服务。
3.网易系应用包含： 网易系非游戏应用：网易云音乐、网易新闻、网易公开 课、网易云阅读、网易云课堂、易信、网易蜗牛阅读、网易 100 分。网易系游戏应用包含：梦幻西游、大话西游、阴阳师、 荒野行动、终结者 2、倩女幽魂、率土之滨、镇魔曲、三国 如龙传、坦克连、决战平安京、我的世界、天下、光明大陆、 格罗娅传奇。
4.阿里系定向流量包含：优酷视频
5.腾讯视频 所有 APP“专属流量”使用说明 （一）在电信网络下，使用对应免流 APP 产生的流量； （二）所使用的 APP 需升级到最新版本； （三）注册所有 APP 手机号码需与本机号码一致。
不免流范围
1、在无线上网卡、移动 WIFI、MIFI、平板电脑（如 ipad）等 设备使用时；
2、将手机号码作为手机热点使用时；
3、使用网络加速器、代理服务器、VPN 等工具时；
4、在国际地区、港澳台使用时；
5、通过 APP STORE 或安卓市场等下载/更新应用时；
6、应用内开机广告及第三方广告时产生的流量；
7、应用中访问外部链接或转发到外部应用产生的流量；
8、APP 内直播业务均不免流；
9、所有 APP 内的第三方提供的网页（手机百度搜索跳转的第 三方网页除外）视频、图片、广告、应用下载等服务，例如手 机新闻网、搜狗、携程、京东等非百度业务。
购买须知
1.本套餐一个证件只能办理一张，发货后请使用订购人身份证 原件按照随卡寄送的单页说明进行激活，若发货后超过 20 天 未激活，系统将回收处理。
2.根据国家相关规定，未满 16 周岁的用户将不能在网络渠道 办理入网，因年龄问题未能激活的用户，请由您的法定代理人 前往营业厅代为办理，详情请咨询 10000.
3.本号卡仅满足用户正常使用通话和流量，对于因受业务及物 流限制地区可能产生电信诈骗的归属地我们会进行鉴别并不 支持发货，敬请见谅。
4.未满 16 周岁，一证五号已满（含正在使用的）国政通不通 过，军人，保密单位等职业的身份证属于特殊证件无法网上办 理通讯业务，故不发货。`