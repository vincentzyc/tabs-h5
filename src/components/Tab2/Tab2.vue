<template>
  <div class="form-wrapper">
    <BaseForm @submit="submitOrder" />
  </div>
  <Agreement title="填写并提交视为阅读并同意" :agrList="agrList" v-model:checked="checked" />
  <img src="@/assets/img/tab2-2.jpg" alt="产品说明" class="widthfull" />
  <img src="@/assets/img/tab2-3.jpg" alt="产品说明" class="widthfull" />
  <div class="text-center expenses-wrap">
    <span class="expenses-text" @click="showExpenses()">资费详情说明</span>
  </div>
  <van-popup class="agreement-popup" v-model:show="show" round>
    <div class="agreement-wrap">{{ expensesText }}</div>
    <div class="agreement-confirm" @click="close()">我知道了</div>
  </van-popup>
</template>

<script setup lang="ts">
import { expensesText } from "./static-data";

provide("tabIndex", 1);

let show = $ref(false),
  checked = $ref(true),
  agrList = $ref([
    {
      title: "《个人信息保护政策》",
      text: "https://h5.lipush.com/h5/index.html?id=5523261663728019648",
    },
    {
      title: "《入网许可协议》",
      text: "https://h5.lipush.com/h5/index.html?id=4554931263728179465",
    },
    {
      title: "《单独同意书》",
      text: "https://h5.lipush.com/h5/index.html?id=7883210563728253535",
    },
  ]);
function showExpenses() {
  show = true;
}
function close() {
  show = false;
}
function submitOrder() {
  checked = true;
}
</script>
