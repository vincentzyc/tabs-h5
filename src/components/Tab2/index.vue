<template>
  <div>电信</div>
  <div class="text-center expenses-wrap">
    <span class="expenses-text" @click="showExpenses()">资费详情说明</span>
  </div>
  <van-popup class="center-popup" v-model:show="show" round>
    <div class="popup-wrap">{{expensesText}}</div>
  </van-popup>
</template>

<script setup lang="ts">
import { expensesText } from "./static-data";
let show = $ref(false);
function showExpenses() {
  show = true;
}
</script>
