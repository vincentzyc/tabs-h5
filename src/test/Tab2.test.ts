
import { mount } from '@vue/test-utils'
import { Lazyload } from 'vant';
import { createTestingPinia } from '@pinia/testing'
import Tab2 from '@/components/Tab2/Tab2.vue'

test('mount Tab2 component', async () => {
  expect(Tab2).toBeTruthy()

  const wrapper = mount(Tab2, {
    global: {
      plugins: [createTestingPinia()],
      directives: {
        lazy: Lazyload as any
      }
    },
  })
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.get('.expenses-text').trigger('click')

  await nextTick()

  expect(wrapper.find('.agreement-popup').exists()).toBe(true)
})
