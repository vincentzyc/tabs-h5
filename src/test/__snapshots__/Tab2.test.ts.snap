// Vitest Snapshot v1

exports[`mount Tab2 component 1`] = `
"<div class=\\"form-wrapper\\">
  <div class=\\"form-wrap\\">
    <div class=\\"flex align-middle form-item\\" id=\\"custName\\">
      <div class=\\"form-label\\">领卡姓名</div>
      <div class=\\"form-input flex flex-auto align-middle\\"><input maxlength=\\"15\\" placeholder=\\"输入办理人姓名（已加密）\\" type=\\"text\\"></div>
    </div>
    <div class=\\"flex align-middle form-item\\" id=\\"contactNumber\\">
      <div class=\\"form-label\\">联系电话</div>
      <div class=\\"form-input flex-auto flex align-middle\\"><input maxlength=\\"11\\" placeholder=\\"输入有效联系电话并保持畅通\\" type=\\"tel\\"></div>
    </div>
    <div style=\\"display: none;\\">
      <div class=\\"flex align-middle form-item\\" id=\\"idCardNo\\">
        <div class=\\"form-label\\">身份证号</div>
        <div class=\\"form-input flex-auto flex align-middle\\"><input maxlength=\\"18\\" placeholder=\\"输入办理人身份证（已加密）\\" type=\\"text\\"></div>
      </div>
      <!---->
      <div class=\\"flex align-middle form-item border-bnone\\" id=\\"address\\">
        <div class=\\"form-label\\">详细地址</div>
        <div class=\\"form-input flex-auto flex align-middle\\"><input maxlength=\\"50\\" placeholder=\\"街道/镇+村/小区/楼+门牌号\\" type=\\"text\\"></div>
      </div>
    </div><img src=\\"/src/assets/img/button.png\\" alt=\\"提交按钮\\" class=\\"breathlamp widthfull\\">
  </div>
</div>
<div class=\\"marquee-single flex mg-b10\\">
  <ul class=\\"scroll-wrapper\\">
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none man\\"></span><span class=\\"scroll-text txtover\\">李**抢得靓号 1305678****</span></li>
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none man\\"></span><span class=\\"scroll-text txtover\\">蔡**抢得靓号 1308288****</span></li>
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none woman\\"></span><span class=\\"scroll-text txtover\\">张**抢得靓号 1326668****</span></li>
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none woman\\"></span><span class=\\"scroll-text txtover\\">罗**抢得靓号 1665688****</span></li>
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none man\\"></span><span class=\\"scroll-text txtover\\">林**抢得靓号 1556688****</span></li>
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none woman\\"></span><span class=\\"scroll-text txtover\\">周**抢得靓号 1306566****</span></li>
  </ul>
</div>
<div role=\\"checkbox\\" class=\\"van-checkbox van-checkbox--label-disabled\\" tabindex=\\"0\\" aria-checked=\\"true\\">
  <div class=\\"van-checkbox__icon van-checkbox__icon--round van-checkbox__icon--checked\\" style=\\"font-size: 14px;\\"><i class=\\"van-badge__wrapper van-icon van-icon-success\\" style=\\"border-color: #cccccc; background-color: rgb(204, 204, 204);\\">
      <!---->
      <!---->
      <!---->
    </i></div><span class=\\"van-checkbox__label\\"><p class=\\"agreement-text\\"><span>填写并提交视为阅读并同意</span><span><!--v-if--><span class=\\"agreement-title\\">《个人信息保护政策》</span></span><span><span>和</span><span class=\\"agreement-title\\">《入网许可协议》</span></span><span><span>和</span><span class=\\"agreement-title\\">《单独同意书》</span></span></p></span>
</div>
<!--teleport start-->
<!--teleport end-->
<img src=\\"/src/assets/img/tab2-2.jpg\\" alt=\\"产品说明\\" class=\\"widthfull\\">
<img src=\\"/src/assets/img/tab2-3.jpg\\" alt=\\"产品说明\\" class=\\"widthfull\\">
<div class=\\"text-center expenses-wrap\\"><span class=\\"expenses-text\\">资费详情说明</span></div>
<transition-stub name=\\"van-fade\\" appear=\\"true\\" persisted=\\"false\\" css=\\"true\\" role=\\"button\\" tabindex=\\"0\\">
  <!---->
</transition-stub>
<transition-stub name=\\"van-fade\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
  <!---->
</transition-stub>"
`;
